apps/sms/views/shared/test/unit/contacts_test.js

```diff
@@ -262,169 +121,130 @@ suite('Contacts', function(done) {
-
-        // navigator.mozContacts.find was called?
-        assert.equal(mHistory.length, 1);
       }).then(done, done);
     });
 
 
     test('predominate first, lower', function(done) {
-      var mozContacts = navigator.mozContacts;
+      var result = MockContact.list();
 
-      Contacts.findContactByString('pepi o').then(function(contacts) {
-        var mHistory = mozContacts.mHistory;
+      navigator.mozContacts.find.returns(Promise.resolve(result));
 
+      Contacts.findContactByString('pepi o').then(function(contacts) {
         // No contacts were found
         assert.ok(Array.isArray(contacts));
         assert.equal(contacts.length, 1);
-
-        // navigator.mozContacts.find was called?
-        assert.equal(mHistory.length, 1);
       }).then(done, done);
     });
 
     test('predominate last, lower', function(done) {
-      var mozContacts = navigator.mozContacts;
+      var result = MockContact.list();
 
-      Contacts.findContactByString('o pepi').then(function(contacts) {
-        var mHistory = mozContacts.mHistory;
+      navigator.mozContacts.find.returns(Promise.resolve(result));
 
+      Contacts.findContactByString('o pepi').then(function(contacts) {
         // No contacts were found
         assert.ok(Array.isArray(contacts));
         assert.equal(contacts.length, 1);
-
-        // navigator.mozContacts.find was called?
-        assert.equal(mHistory.length, 1);
       }).then(done, done);
     });
 
     test('multi-word name', function(done) {
-      var mozContacts = navigator.mozContacts;
+      var result = MockContact.list([
+        { givenName: ['Mary Anne'], familyName: ['Jones'] }
+        ]);
 
-      Contacts.findContactByString('mary anne').then(function(contacts) {
-        var mHistory = mozContacts.mHistory;
+      navigator.mozContacts.find.returns(Promise.resolve(result));
 
+      Contacts.findContactByString('mary anne').then(function(contacts) {
         // No contacts were found
         assert.ok(Array.isArray(contacts));
         assert.equal(contacts.length, 1);
-
-        // navigator.mozContacts.find was called?
-        assert.equal(mHistory.length, 1);
       }).then(done, done);
     });
 
     test('name first, part of tel number last', function(done) {
-      var mozContacts = navigator.mozContacts;
+      var result = MockContact.list();
 
-      Contacts.findContactByString('Pepito 8888').then(function(contacts) {
-        var mHistory = mozContacts.mHistory;
+      navigator.mozContacts.find.returns(Promise.resolve(result));
 
+      Contacts.findContactByString('Pepito 8888').then(function(contacts) {
         // contacts were found
         assert.ok(Array.isArray(contacts));
         assert.equal(contacts.length, 1);
-
-        // navigator.mozContacts.find was called?
-        assert.equal(mHistory.length, 1);
       }).then(done, done);
     });
 
     test('part of tel number first, name last', function(done) {
-      var mozContacts = navigator.mozContacts;
```

Comment:
ah yes I get it; I missed you were calling restore in suiteTeardown and not just teardown :) this is not a good practice; if you need to regroup the same stubs definition, it's better to do them in a setup() for this suite (or maybe extract a separate suite if it's not existing and a lot of cases are similar)